### ISSUE TITLE
watch-status-change.py example script doesn't actually receive any events

### ISSUE THREAD

**Issue by jfly** [OpenVPN/openvpn3-linux#197]
I was trying to run the [watch-status-change.py example script](https://github.com/OpenVPN/openvpn3-linux/blob/v20/src/tests/python/watch-status-change.py), and it simply doesn't print any messages at all.

After a lot of futzing around, I eventually discovered that it works if I register a `LogCallback` in addition to [the `StatusChangeCallback` it already has](https://github.com/OpenVPN/openvpn3-linux/blob/v20/src/tests/python/watch-status-change.py#L45).

Then some searching lead me to [this comment](https://github.com/OpenVPN/openvpn3-linux/issues/129#issuecomment-1159251468) from @dsommers:

> With v18_beta several signals are no longer broadcasted as it was before.
> [...]
> You need to call the [net.openvpn.v3.sessions.LogForward](https://github.com/OpenVPN/openvpn3-linux/blob/master/docs/dbus/dbus-service-net.openvpn.v3.sessions.md#method-netopenvpnv3sessionslogforward) method with the true boolean to enable Log and StatusChange events

Which explains why this script isn't working. It was probably written before v18_beta and simply never updated.

I see 2 ways of fixing this:

1. Change watch-status-change.py to register a noop `LogCallback`.
2. Or, change [`StatusChangeCallback`](https://github.com/OpenVPN/openvpn3-linux/blob/v20/src/python/openvpn3/SessionManager.py#L312) to also call `LogForward` as `LogCallback` does (I think this would require a little bit of bookkeeping to implement correctly, because if you have both callbacks registered, you can't call `LogForward(False)` until both callbacks are removed).

I personally feel like 2) is a better option. If this sounds reasonable, I'd be happy to put together a PR.

**Comment by dsommers** [comment]
On a holiday now, just a quick reply. You are spot on.  The v18_beta release did definitely break this example script.

After having had a quick look at the `LogForward` method in `sessionmgr.hpp`, some bookkeeping is indeed needed.  I believe the sessionmgr.hpp should actually implement a duplication check as well; which is outside the scope of this issue - but a potential side-effect of failed bookkeeping.

If you're willing to put some efforts into a pull-req for 2), I'd be delighted to review it and pull in your improvement.

**Comment by jfly** [comment]
Sweet! That was my first time sending a patch by email. Hope you got it!

**Comment by dsommers** [comment]
@jfly I've seen your patch to the ML, that looks good.  I'll give it a good look when I'm back from holiday.

https://www.mail-archive.com/openvpn-devel@lists.sourceforge.net/msg26840.html
Message-id: <20230709231929.195048-1-jeremyfleischman@gmail.com>

**Comment by jfly** [comment]
@dsommers, quick ping re: this change I sent into the ML. Have you had a chance to take a look at it yet?

**Comment by dsommers** [comment]
@jfly Thanks for the ping! I'm catching up after the holiday.  Your patch is not forgotten!

**Comment by dsommers** [comment]
Copy of the message just sent to the [mailing list](https://www.mail-archive.com/openvpn-devel@lists.sourceforge.net/msg26999.html)
```

Thanks a lot for your patience and persistence through this review.

Your patch has been put into the queue for the coming v21 release. I've not yet
pushed out that branch, as I'm waiting for our QA team to complete the testing
of an updated OpenVPN 3 Core library release first; which is required to be able
to build OpenVPN 3 Linux from the public repos.


But consider your patch applied. I've only done minor edits to some comments and
commit messages, but the code itself is unchanged.


I'll follow-up with an update once this commit is public.
```

**Comment by dsommers** [comment]
Just pushed out a `releaseprep/v21` branch wit this fix included; as commit ba6fe37e7e28d1e633b56052383da3072f03c11e.  Once QA has tested this release preparation branch, it will get merged to master.
